graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -213,28 +190,47 @@ class VertexType(GraphElement):
     def __init__(self, class_name, abstract, properties, class_fields):
         super(VertexType, self).__init__(class_name, abstract, properties, class_fields)
 
-        # Non-edges must not have properties like "in" or "out" defined, and
-        # must not have properties of type "Link".
-        _validate_non_edges_do_not_have_edge_like_properties(class_name, properties)
-
 
 class EdgeType(GraphElement):
-    def __init__(self, class_name, abstract, properties, class_fields):
-        super(EdgeType, self).__init__(class_name, abstract, properties, class_fields)
+    EDGE_SOURCE_PROPERTY_NAME = 'out'
+    EDGE_DESTINATION_PROPERTY_NAME = 'in'
+    EDGE_END_NAMES = {EDGE_SOURCE_PROPERTY_NAME, EDGE_DESTINATION_PROPERTY_NAME}
+
+    def __init__(self, class_name, abstract, properties, class_fields, base_connections):
+        """Create a new EdgeType object.
+
+        Args:
+            class_name: string, the name of the schema element class.
+            abstract: bool, True if the class is abstract, and False otherwise.
+            properties: dict, property name -> PropertyDescriptor describing the properties of
+                        the schema element.
+            class_fields: dict, class field name -> class field value, both strings
+            base_connections: dict, string -> string. Maps a edge end, (one of 'in' or 'out'),
+                              to the class that is allowed at that edge end and is the superclass of
+                              all the classes allowed in that edge end. A base connection may
+                              be missing for either end if the edge is abstract.
+
+        Returns:
+            a EdgeType with the given parameters
+        """
```

Comment:
I am not sure how to write docs for one new field, (base_connections), so I added documentation for all the fields.